src/AccessibilityInsights.SharedUx/ViewModels/EventConfigNodeViewModel.cs

```diff
@@ -22,47 +22,117 @@ namespace AccessibilityInsights.SharedUx.ViewModels
     public class EventConfigNodeViewModel : ViewModelBase, INotifyPropertyChanged
     {
         /// <summary>
-        /// Child ViewModels
+        /// Child ViewModels - external code can read but not modify
         /// </summary>
-        public ObservableCollection<EventConfigNodeViewModel> Children { get; private set; }
+        public IReadOnlyCollection<EventConfigNodeViewModel> Children => this._children;
+
+        private readonly ObservableCollection<EventConfigNodeViewModel> _children;
 
         /// <summary>
         /// Depth for treeviewitem
         /// </summary>
         public int Depth { get; set; }
 
         /// <summary>
-        /// Is node checked
+        /// Status of IsThreeState property
+        /// </summary>
+        public bool IsThreeState { get; } = false;
+
+        /// <summary>
+        /// Is node checked (null if in "mixed" ThreeState value)
         /// </summary>
-        private bool _ischecked = false;
-        public bool IsChecked
+        private bool? _ischecked = false;
+        public bool? IsChecked
         {
             get
             {
-                return this._ischecked;
+                return this.IsThreeState ? this._ischecked :
```

Comment:
To me this feels easier to parse w/o the nested ternary, would something like `
return this.isThreeState? this._ischecked : this._ischecked == true` work?

Reply:
Missed this before merging. It's in #890